tensorboard/components/tensor_widget/tensor-widget-impl.ts

```diff
@@ -526,18 +543,28 @@ export class TensorWidgetImpl implements TensorWidget {
           this.renderSelection();
         });
         valueDiv.addEventListener('mouseenter', () => {
-          const detailedValue = valueDiv.getAttribute(DETAILED_VALUE_ATTR_KEY);
-          if (!detailedValue) {
+          let detailedValue:
+            | number
+            | boolean
+            | string
+            | null = valueDiv.getAttribute(DETAILED_VALUE_ATTR_KEY);
+          const valueClass = this.getValueClass();
+          if (valueClass !== 'string' && !detailedValue) {
```

Comment:
OK, but now it's even more confusing because of the `JSON.parse()`, which IIUC will convert `"true"` and `"false"` to the boolean values `true` and `false`, thereby actually manifesting the issue I was worried about in the first place.  Similarly a detailedValue of the number `0` will now do the wrong thing, I think.

Aha: you already returned if `!valueAttr`, which I think is all you need anyway.  Can you delete this conditional return entirely?